### PR TITLE
typography development

### PR DESCRIPTION
**Pull request by eraymeric**

### CODE REVIEW

File: src/components/typography/typography.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import React, { HTMLAttributes } from 'react'
+import styled from 'styled-components'
+
+const colors = {
```
Comment (usmaniqbal998) [inline src/components/typography/typography.tsx:4]: lets use global variables defined in scss for colors here

File: src/components/typography/typography.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  primary: '#18181B',
+  secondary: '#71717A',
+}
+
+const variants = {
+  heading2XL: {
+    fontSize: '4rem',
```
Comment (usmaniqbal998) [inline src/components/typography/typography.tsx:11]: can you write a callback for this repetitive code like following 
```

function variantMixin(fontSize: string, fontWeight: number, lineHeight: string, letterSpacing: string) {
  return {
    ['font-size']: fontSize,
    ['font-weight']: fontWeight,
    ['line-height']: lineHeight,
    ['letter-spacing']: letterSpacing,
  }
}
```

and then all you have to do for every variant is 
```
 heading2XL: {
    ...variantMixin('4rem', 600, '4rem', '0%'),
  },
```

File: src/components/typography/typography.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+export interface TypographyProps extends HTMLAttributes<HTMLParagraphElement> {
+  variant?: keyof typeof variants
+  color?: keyof typeof colors
+  size?: string
+  children: React.ReactNode
+  as?: TypographyElements
+}
+
+const TypographyElement = styled.p<TypographyProps>`
+  font-family: 'Satoshi', sans-serif;
```
Comment (usmaniqbal998) [inline src/components/typography/typography.tsx:137]: is this font family bundled with our package ?

**Review by usmaniqbal998 — CHANGES_REQUESTED** [review]

File: src/components/typography/typography.tsx (hunk 3 above)
Comment (birkanalp) [reply src/components/typography/typography.tsx:137]: I think we should import font families in our projects

**Review by birkanalp — APPROVED** [review]

**Review by usmaniqbal998 — APPROVED** [review]

**Pull request merged**